makefile

```diff
@@ -4,6 +4,7 @@ NASM ?= nasm
 ARCHFLAG ?= -m64
 DFLAGS = $(ARCHFLAG) -w -debug -g -unittest
 PLATFORM = $(shell uname -s)
+LD_GOLD = $(shell which ld.gold | xargs basename)
```

Comment:
Check this only on linux. Gold only support ELF, which doesn't work on OSX anyway.